icu4j/main/classes/core/src/com/ibm/icu/impl/number/parse/IgnorablesMatcher.java

```diff
@@ -18,6 +18,9 @@
     public static final IgnorablesMatcher STRICT = new IgnorablesMatcher(
             StaticUnicodeSets.get(StaticUnicodeSets.Key.STRICT_IGNORABLES));
 
+    public static final IgnorablesMatcher COMPATIBILITY = new IgnorablesMatcher(
+            StaticUnicodeSets.get(StaticUnicodeSets.Key.EMPTY));
```

Comment:
This is indeed very strict.  JDK does not allow any ignorables, not even bidi marks?  This could cause issues with things like plus and minus signs from data that have ALMs.

Reply:
yes, OpenJDK does not have ignorables. Added NumberFormatJavaCompatilityTest. PTAL